.circleci/mkci

```diff
@@ -0,0 +1,177 @@
+            beginning = commits[0:8]
+            end = commits[43:43+8]
+            return f"{base}/{beginning}...{end}"
+
+    return url
+
+
+
+def buildpkg(p) -> List[Path]:
+    """Build one package, installing required dependencies"""
+    print(f"Building package in {p}")
+    ver = run2("dpkg-parsechangelog --show-field Version", cwd=p).strip()
+    assert ver, f"No version number found in {p}/debian/changelog"
+    sudo = True
+    if getenv("CIRCLECI"):
+        # Running in CircleCI
+        sudo = False
+        comp = trim_compare(getenv("CIRCLE_COMPARE_URL", ""))  # show changes in VCS
+        if getenv("CIRCLE_PULL_REQUEST"):
+            # This is a PR: build ~pr<N>-<N> version. CIRCLE_PR_NUMBER is broken
+            pr_num = getenv("CIRCLE_PULL_REQUEST", "").rsplit("/", 1)[-1]
+            build_num = getenv("CIRCLE_BUILD_NUM")
+            ver = f"{ver}~pr{pr_num}-{build_num}"
+            run2(f"dch -b -v {ver} {comp}", cwd=p)
+            run2(f"dch -r {ver} {comp}", cwd=p)
+            ver2 = run2("dpkg-parsechangelog --show-field Version", cwd=p).strip()
+            assert ver == ver2, ver + " <--> " + ver2
+        else:
+            run2(f"dch -b -v {ver} {comp}", cwd=p)
+            run2(f"dch -r {ver} {comp}", cwd=p)
+            ver2 = run2("dpkg-parsechangelog --show-field Version", cwd=p).strip()
+            assert ver == ver2, ver + " <--> " + ver2
+
+    runi("apt-get build-dep -qy --no-install-recommends .", p, sudo=sudo)
+    runi("fakeroot debian/rules build", p)
+    runi("fakeroot debian/rules binary", p)
+    with p.joinpath("debian/files").open() as f:
+        return [p.parent.joinpath(line.split()[0]) for line in f]
+
+
+def ci(args) -> None:
+    # TODO: detect sudo presence
+    err = "BINTRAY_USERNAME or BINTRAY_API_KEY environment variables not set"
+    assert getenv("BINTRAY_USERNAME") and getenv("BINTRAY_API_KEY"), err
+    run2("apt-get update -q")
+    run2("apt-get install -qy --no-install-recommends git")
+    pkgdirs = detect_changed_packages()
+    if not pkgdirs:
+        return
+    print(f"Building {pkgdirs}")
+    run2("apt-get install -qy --no-install-recommends devscripts")
+
+    pkgs_lists = [buildpkg(pd) for pd in pkgdirs]
+    for pli in pkgs_lists:
+        for p in pli:
+            upload(p, args)
+
+def build() -> None:
+    """Run manual build on workstation"""
+    pkgdirs = detect_changed_packages()
+    pkgs_lists = [buildpkg(pd) for pd in pkgdirs]
+    print("Outputs:")
+    for pli in pkgs_lists:
+        for p in pli:
+            print(p)
+
+
+def upload(fi, args) -> None:
+    """Upload to Bintray"""
+    assert fi.is_file()
+    pname, pver, arch = fi.name.split("_")
+    btuser = getenv("BINTRAY_USERNAME")
+    assert btuser
+    auth = HTTPBasicAuth(btuser, getenv("BINTRAY_API_KEY"))
+    url = f"{BINTRAY_API}/content/{args.org}/{args.repo}/{pname}/{pver}/{fi.name};deb_distribution=unstable;deb_component=main;deb_architecture=amd64;publish=1"
```

Comment:
I wonder if you want to maybe add a version check at the beginning of the script to ensure it's python > 3.7

Reply:
Also maybe wrap this line

Reply:
Done